src/neo-vm/ExecutionEngine.cs

```diff
@@ -1193,15 +992,44 @@ protected virtual void LoadContext(ExecutionContext context)
 
         public ExecutionContext LoadScript(Script script, int rvcount = -1)
         {
-            ExecutionContext context = new ExecutionContext(script, CurrentContext?.Script, rvcount);
+            ExecutionContext context = new ExecutionContext(script, CurrentContext?.Script, rvcount, ReferenceCounter);
             LoadContext(context);
             return context;
         }
 
         protected virtual bool OnSysCall(uint method) => false;
 
-        protected virtual bool PostExecuteInstruction(Instruction instruction) => true;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public StackItem Peek(int index = 0)
+        {
+            var stack = CurrentContext?.EvaluationStack ?? ResultStack;
+            return stack.Peek(index);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public StackItem Pop()
+        {
+            return CurrentContext.EvaluationStack.Pop();
```

Comment:
I prefer to use the `context` variable in `ExecuteInstruction` because here you need to compute again witch is the 
`CurrentContext`

Reply:
Optimized.